openfl/federated/data/sources/data_sources_json_parser.py

```diff
@@ -31,48 +33,65 @@ def parse(json_string: str) -> VerifiableDatasetInfo:
         except json.JSONDecodeError as e:
             raise ValueError(f"Invalid JSON format: {e}")
 
-        datasources = DataSourcesJsonParser.process_data_sources(data)
+        datasources = DataSourcesJsonParser.process_data_sources(data, check_dir_traversal)
         if not datasources:
             raise ValueError("No data sources were found.")
         return VerifiableDatasetInfo(
             data_sources=datasources,
-            label="",
+            label=label,
+            metadata=metadata,
         )
 
     @staticmethod
-    def process_data_sources(data):
+    def process_data_sources(data, check_dir_traversal=False):
         """Process and validate data sources."""
-        cwd = os.getcwd()
+        os.getcwd()
```

Comment:
This only returns `cwd`. If `cwd` is unused, line also can be removed.

Reply:
sorry right. removed of course